tools/cov2json/ProfileData.cpp

```diff
@@ -0,0 +1,182 @@
+  if (file.size()) {
+    std::error_code error;
+    auto os = make_unique<raw_fd_ostream>(file, error, sys::fs::F_RW);
+    if (error)
+      exitWithErrorCode(error, "opening output stream");
+    return move(os);
+  } else {
+    return make_unique<raw_fd_ostream>(fileno(stdout),
+                                       /* shouldClose = */ false);
+  }
+}
+
+/// Creates a Function struct off of a FunctionRecord.
+Function::Function(const coverage::FunctionRecord &record) {
+  this->symbol = extractSymbol(record.Name);
+  std::string demangled;
+  getDemangled(symbol, demangled);
+  this->name = demangled;
+  for (auto &region : record.CountedRegions) {
+    if (region.FileID != region.ExpandedFileID)
+      continue;
+    if (region.Kind !=
+        llvm::coverage::CounterMappingRegion::RegionKind::CodeRegion)
+      continue;
+    Region r(region.ColumnStart, region.ColumnEnd, region.LineStart,
+             region.LineEnd, region.ExecutionCount);
+    regions.emplace_back(r);
+  }
+  executionCount = record.ExecutionCount;
+}
+ 
+uint64_t Function::regionCount() {
+  return regions.size();
+}
+
+uint64_t Function::executedRegionCount() {
+  uint64_t count = 0;
+  for (auto &region : regions) {
+    if (region.executionCount > 0) {
+      count += 1;
+    }
+  }
+  return count;
+}
+  
+uint64_t File::regionCount() {
+  size_t counter = 0;
+  for (auto &function : functions) {
+    counter += function.regionCount();
+  }
+  return counter;
+}
+
+uint64_t File::executedRegionCount() {
+  size_t counter = 0;
+  for (auto &function : functions) {
+    counter += function.executedRegionCount();
+  }
+  return counter;
+}
+
+/// Loads a CoverageMapping for a CoverageFilePair
+std::unique_ptr<llvm::coverage::CoverageMapping>
+CoverageFilePair::coverageMapping() {
+  auto map = llvm::coverage::CoverageMapping::load(binary, filename);
+  
+  if (auto error = map.getError()) {
+    exitWithErrorCode(error, "loading coverage map");
+  }
+  
+  return move(map.get());
+}
+
+/// Loads a list of files that are within the provided coveredDir.
+void CoverageFilePair::loadFileMap(std::vector<File> &files,
+                                   StringRef coveredDir,
+                                   std::vector<std::string> &coveredFiles) {
+  auto mapping = coverageMapping();
+  for (auto &filename : mapping->getUniqueSourceFiles()) {
+    StringRef truncatedFilename = filename;
+    if (coveredDir != "") {
+      if (!filename.startswith(coveredDir)) {
+        continue;
+      }
+      StringRef parentPath = sys::path::parent_path(coveredDir);
+      truncatedFilename = filename.substr(parentPath.size(),
+                                          filename.size() - parentPath.size());
+      truncatedFilename = sys::path::relative_path(truncatedFilename);
+    }
+    if (!coveredFiles.empty()) {
+      bool found = false;
+      for (auto &file : coveredFiles) {
+        if (truncatedFilename.endswith(file)) {
```

Comment:
Hmm...I hadn't considered that. The coveredDir is used to, say, eliminate the common `/Users/name/path/to/Swift/` prefix. The `file` here will be `swift/lib/Expr.cpp`, for example.

Do you think that's a common enough case, that there are two files with the same suffix in the project?